### PR TITLE
Make singleheader tests be test-only

### PR DESCRIPTION
**Pull request by jkeiser**
This addresses a comment from #962 that singleheader tests don't really need to be a part of the default compile.

### CODE REVIEW

File: singleheader/CMakeLists.txt [hunk 1]
```diff
@@ -113,8 +115,9 @@ add_test(amalgamate_demo amalgamate_demo ${EXAMPLE_JSON} ${EXAMPLE_NDJSON})
 
 install(FILES simdjson.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-
-
+add_library(simdjson-singleheader STATIC "")
```
Comment (lemire) [inline singleheader/CMakeLists.txt:118]: Shouldn't this be labelled with EXCLUDE_FROM_ALL.

I am concerned that this thing could get built and installed.

Comment (jkeiser) [reply singleheader/CMakeLists.txt:118]: @lemire that's what add_compile_only_test() does. Hmm. Trying to figure out if there's a better name ... suggestions welcome.

Comment (lemire) [reply singleheader/CMakeLists.txt:118]: I verified for myself that it does not get installed.

Comment (lemire) [reply singleheader/CMakeLists.txt:118]: I guess I just don't understand what "add_library" does.  The documentation says...

```
Adds a library target called <name> to be built from the source files listed in the command invocation. (The source
 files can be omitted here if they are added later using target_sources().) The <name> corresponds to the logical
 target name and must be globally unique within a project. The actual file name of the library built is constructed
 based on conventions of the native platform (such as lib<name>.a or <name>.lib).

With EXCLUDE_FROM_ALL set to false or not set at all, the target will be brought up to date as part of doing a make
 install or its equivalent for the CMake generator being used.
```

Comment (lemire) [reply singleheader/CMakeLists.txt:118]: Oh. I see that you patch it up later...

```
    set_target_properties(${TEST_NAME} PROPERTIES EXCLUDE_FROM_ALL TRUE EXCLUDE_FROM_DEFAULT_BUILD TRUE)

```

Good.

**Pull request merged**